scripts/prepare_firmware_binaries.py

```diff
@@ -57,6 +60,30 @@ def main() -> int:
             raise RuntimeError(f"Missing {arch} directory in {in_path}")
         shutil.copytree(in_path / arch, pathlib.Path(tmp_dir.name), dirs_exist_ok=True)
 
+        tmp_path = pathlib.Path(tmp_dir.name)
+        for bin_file in tmp_path.rglob("*.bin"):
+            receipt = {}
+
+            try:
```

Comment:
This nested try except hurts lmao

Can you do something along the lines of this?

``` python
def get_receipt(bin_file):
    for method in (get_unsigned_payload_receipt, get_signed_payload_receipt):
        try:
            return method(bin_file)
        except Exception as e:
            pass
    
    logging.warning(f"Failed to get receipt for {bin_file}")
    raise ValueError("Failed to get receipt")
```